timemachine/lib/custom_ops.py

```diff
@@ -1,6 +1,14 @@
-# (ytz): this file is left intentionally blank.
-# it's a trick to allow us to do a dummy import on the client side
-# that don't have custom_ops shared objects.
+# This file contains stubs for some classes defined in the C++
+# extension module.
+#
+# The purpose is to allow importing modules with
+# unavoidable top-level references to objects defined in the extension
+# (e.g. modules containing subclasses of classes defined in the C++
+# code).
+#
+# If the extension module .so file is present, the definitions
+# in it will take precedence over the stubs defined here.
 
-# on the worker side, a custom .so file for the custom_op and will
-# have higher import priority than the .py file
+
+class FanoutSummedPotential:
```

Comment:
What is the reason to inherit from this and not construct it in the same way that we do other potentials? I find inheriting from C++ objects sketchy, as I don't understand the mechanisms underlying it.

Reply:
My reasoning is that `NonbondedImplWrapper` should have the same interface as the C++ `Potential` class (because it's returned by `Nonbonded.unbound_impl()`; i.e. it's not a `CustomOpWrapper`). The motivation for this was to allow `Nonbonded` as implemented as a `FanoutSummedPotential` to maintain the same interface with `disable_hilbert_sort()` and `set_nblist_padding()` -- the wrapper class adds these methods to the underlying `FanoutSummedPotential` class.

I agree that this is a bit hacky and confusing. One alternative could be to implement something like `NonbondedImplWrapper` on the C++ side. I preferred the current approach because it doesn't require an ad-hoc C++ class for each variant of the nonbonded potential. Longer term, it might be cleaner to change the API so that e.g. `disable_hilbert_sort()` isn't called directly on the full nonbonded potential (which is typically a sum of parts), but on the underlying all pairs or interaction group potential.

Reply:
Add this as a comment to `NonbondedImplWrapper`.